text/0240-app-telemetry-schema-vision-roadmap.md

```diff
@@ -0,0 +1,370 @@
+policies in parallel, knowing that the policies are shared and the entire
+environment is controlled by the same entity.
+
+In the second model, the application/service environment does not have access to
+and does not control the telemetry schema policies applied to the telemetry
+schemas of the libraries. Two approaches for recovering telemetry schemas are
+currently being considered. A specific OTEP will be responsible for identifying
+the best approach.
+
+![Development Strategies to Support](./img/0240-dev-strategies.png)
+
+#### Decentralized Approach
+
+Library designers using this schema-first telemetry schema approach add a phase
+in their CI/CD pipeline for the registration of the resolved telemetry schema
+associated with their library for a given version. This registration is done
+towards a registry maintained by OpenTelemetry.
+
+Application/service owners configure their CI/CD pipeline to automatically
+import the resolved telemetry schemas of dependencies (if they exist) in order
+to apply locally the telemetry schema policies or any other build step that
+applies to/feeds off the resolved telemetry schemas of the dependencies.
+
+![Decentralized Approach](./img/0240-decentralized-approach.png)
+
+This approach is considered decentralized as it does not impose any direct
+collaboration between parties while ensuring consistency between versions of
+dependencies and their associated telemetry schemas.
+
+#### Centralized Approach
+
+In this approach, the library designers have no interaction with the external
+world. They define a telemetry schema for their library, version the resolved
+schema in their repository, and release their library as usual.
+
+On the application and service owners' side, they must either make a local copy
+of the resolved telemetry schemas corresponding to their dependencies (similar
+conceptually to the model followed by protobuf and gRPC) or link to the resolved
+telemetry schema for a specific version. In both cases, the synchronization
+between the versions of the library and the resolved telemetry schema is the
+responsibility of the application owner.
+
+![Centralized Approach](./img/0240-centralized-approach.png)
+
+This approach is considered centralized as the application owner has the
+responsibility of maintaining, in a centralized manner locally, all the
+telemetry schemas of the dependencies.
+
+A specific OTEP dedicated to the resolved telemetry schema will be responsible
+for defining the format and structure of the resolved telemetry schema, as well
+as the best approach to collect the resolved telemetry schemas of the different
+dependencies.
+
+## Roadmap
+
+To facilitate the review process and progressively deliver value to the
+OpenTelemetry community, a series of OTEPs and tools are proposed.
+
+### OTEPs
+
+* Telemetry Schema versions (incremental steps):
+  * V1.1.1: Add a section to import one or several Semantic Convention
```

Comment:
I would like to discuss in a separate OTEP the exact structure of the schema files that "import" vs schema files that are called "resolved schemas" in this OTEP. I think it is important that we always publish the resolved schemas that don't require further link resolution to be useful for processing purposes. We don't have to discuss the details here, just wanted to call out that the ability "import" does not necessarily need to be a feature of the published schema files.

Reply:
I see. This is also related to your subsequent comment about considering the current telemetry schema version 1.1.0 as actually being a precursor to what I refer to here as the resolved telemetry schema. I need to go through the entire document again to make the necessary adjustments.

Reply:
> I think it is important that we always publish the resolved schemas

+1

Reply:
I have added an 'Open Questions' section at the end of this OTEP. The purpose is to clearly expose the important questions that will need to be resolved in the following stages/OTEPs. The current OTEP focuses on the overall vision and the major concepts of this initiative. One of the questions is about having one or two types of schema and also includes a format aspect for the schema(s). The point discussed in your comment will be covered in the context of responding to this question in the next OTEP (see roadmap), which I hope to start soon.